### DISCUSSION TITLE
Make it easier to write plugin (make internal helpers public)

### DISCUSSION THREAD

**Discussion by RichardLindhout** [99designs/gqlgen#1945] [Ideas]
### What happened?
I want to rewrite the resolver generation. But when I've copied plugin I need to copy internal code of gqlgen to support it.
```
github.com/web-ridge/gqlgen-sqlboiler/resolver/resolver.go:11:2: use of internal package github.com/99designs/gqlgen/internal/rewrite not allowed
```
### What did you expect?
That I could copy the code from existing plugins and that it works out of the box

### Proposal
Move /internal packages to public

**Comment by vektah** [comment]
I do kind of hate internal. I wish it was much softer, like a lint error you could ignore. 

It's really there to tell you that the package is subject to breaking changes, moreso than most of gqlgen. That package is pretty scrappy and works for the exact user case in that resolver generator but before making it public it would need some more polish.

You best bet right now it's to copy, I'll leave this issue open but I don't see it taking priority.
